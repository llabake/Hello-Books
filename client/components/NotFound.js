import React from 'react';
import { Link } from 'react-router-dom';
import SearchBar from './common/SearchBar';

const NotFound = (props) =>
  <div className="container" style= {{ height: "95vh"}}>
  <div style={{marginLeft: "0px"}}>
    <h4>This is embarassing, something's missing.</h4>
    <b>404: page not found</b>
  </div>
    <div style={{margin: "10 55px 20 35px",}}>
      <SearchBar />
    </div>
    <div >
      <ul style={{"textAlign": "center"}}>
      <li style={{"display": "block"}}><Link to="/">Home</Link>
      </li>
      <li style={{"display": "block"}}><Link to="/allbooks">Books</Link>
      </li>
      <li style={{"display": "block"}}><Link to="/profile">Profile</Link>
      </li>
      </ul>
    </div>
  </div>

export default NotFound;

