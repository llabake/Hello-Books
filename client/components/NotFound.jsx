import React from 'react';
import { Link } from 'react-router-dom';
import SearchBar from './common/SearchBar';

const NotFound = (props) =>
  <div className="container">
    <div style={{ marginLeft: "0px" }} id="not-found">
      <h4>This is embarassing, something's missing.</h4>
      <b>404: page not found</b>
    </div>
    <div>
      <SearchBar />
    </div>
    <div >
      <ul style={{ "textAlign": "center" }}>
        <li style={{ "display": "block" }}><Link to="/">Home</Link>
        </li>
        <li style={{ "display": "block" }}><Link to="/allbooks">Books</Link>
        </li>
        <li style={{ "display": "block" }}><Link to="/profile">Profile</Link>
        </li>
      </ul>
    </div>
  </div>

export default NotFound;

