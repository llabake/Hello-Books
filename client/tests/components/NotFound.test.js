import React from 'react';

import  NotFound  from '../../components/NotFound';


describe('Test suite for 404 page', () => {
    const wrapper = shallow(<NotFound />);
  it('Should render without errors', () => {
    expect(wrapper.length).toEqual(1);
    expect(wrapper.find('#not-found').text()).toMatch('This is embarassing, something\'s missing.404: page not found');
    expect(wrapper.find('ul').length).toBe(1);
    expect(wrapper.find('ul').children().length).toBe(3);
    expect(wrapper.find('Link').length).toBe(3);

  });

  it('should mount correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});