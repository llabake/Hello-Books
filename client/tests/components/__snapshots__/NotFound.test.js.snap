// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for 404 page should mount correctly 1`] = `
<div
  className="container"
>
  <div
    id="not-found"
    style={
      Object {
        "marginLeft": "0px",
      }
    }
  >
    <h4>
      This is embarassing, something's missing.
    </h4>
    <b>
      404: page not found
    </b>
  </div>
  <div>
    <Connect(SearchBar) />
  </div>
  <div>
    <ul
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <li
        style={
          Object {
            "display": "block",
          }
        }
      >
        <Link
          replace={false}
          to="/"
        >
          Home
        </Link>
      </li>
      <li
        style={
          Object {
            "display": "block",
          }
        }
      >
        <Link
          replace={false}
          to="/allbooks"
        >
          Books
        </Link>
      </li>
      <li
        style={
          Object {
            "display": "block",
          }
        }
      >
        <Link
          replace={false}
          to="/profile"
        >
          Profile
        </Link>
      </li>
    </ul>
  </div>
</div>
`;
